client/pdo/client/controller/contract_controller.py

```diff
@@ -242,6 +243,45 @@ def do_sleep(self, args) :
 
         return False
 
+    # -----------------------------------------------------------------
+    def do_parse(self, args) :
+        """parse -- parse a JSON expression and retrieve values
+        """
+        if self.deferred > 0 : return False
+
+        try :
+            pargs = self.__arg_parse__(args)
+
+            parser = argparse.ArgumentParser(prog='parse')
+            parser.add_argument('-q', '--quiet', help='suppress printing the result', action='store_true')
+            parser.add_argument('-e', '--expression', help='json expression to parse', type=str, required=True)
+            parser.add_argument('-p', '--path', help='path to retrieve within the expression', type=str, required=True)
```

Comment:
I'm a bit confused about the `path` argument. I thought it was the json expression that was being parsed, but it seems like what's actually being evaluated is the path. Does `path` point to a python script that does the actual parsing?

Reply:
path as in "JSONPath" (except that we don't have an approved JSON Path library). Basically it is a python expression that is eval'd in the context of the bindings from the dictionary defined in the JSON blob. For example, `parse -s out -p 'a[0]' -e '{"a" : [1, 2, 3]}'` should return 1.

Reply:
Ah! Ok, I wasn't familiar with JSONPath conceptually, but I understand what this is doing now. Would we be able to make this more explicit by saying something like "JSONPath-style path" in the description of the `path` argument, or would we run into trouble because this is our own implementation of a library that hasn't been approved?